[MHV-45980] Remove attachment modal - 59491

vsaleem opened the pull request:
**Note**: 

## Summary

- Added Remove Attachments modal
- This implementation covers improving the [User flows and navigation], and the accessibility issue that - MUST - User should be able to review, edit, or confirm transactional information before submission.
- The Platform recommendation used for this ac was option 2. Add an intermediate confirmation step between a user clicking 'Delete/Remove' and the issue actually being removed.
- Before, a user using a screenreader did not receive any notice that their attachments were deleted after selecting the remove button.
- Now, the screen-reader will report a status that their attachment file has been removed after following the modal prompts.


## Related issue(s)

JIRA:

- jira.devops.va.gov/browse/MHV-45980
- 
Github Issue:

- github.com/https://github.com/department-of-veterans-affairs/va.gov-team/issues/59491

## Testing done


## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._


|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/49e03d54-dfeb-4a29-9eb6-847ec44e617f) 
![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/831d9e11-1e67-4247-876f-1022845a7dc6)|


## What areas of the site does it impact?

*Va.gov - Secure Messaging*

## Acceptance criteria

1) When an attachment is attached to the UI, then the focus should read the file name on the card.
2) The "Remove" button on attachment should have an aria-label that reads, "Remove [file name]".
3) After the Remove button is selected, a modal appears: user confirms remove modal.
4) After user confirms removing the attachment, the attachment card disappears and focus lands on the "Attach File" button, and the screenreader status is announced that "File [file name] successfully removed."

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

Commit:
[MHV-45995] Fix paperclip icon on thread list view

Commit:
Merge branch 'main' into 45995-Resolve-Paperclip-icons-NVDA-SR

Commit:
[MHV-45995] Updated aria-label

- updated label to receive state call to attachments before expanding accordion

Commit:
cleanup

Commit:
Merge branch 'main' into 45995-Resolve-Paperclip-icons-NVDA-SR

Commit:
[MHV-45995] Updated attachment dependency

Commit:
[MHV-45980] Added Remove Attachment  Modal

- added intermediate confirmation step to announce to users that their attachment has been removed.

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-#59491

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-#59491

Commit:
[MHV-45980] Styling Updates

-Styled Remove Attachments modal-button

Commit:
[MHV-45980] Updated unique keys; cleanup

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-#59491

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-#59491

Commit:
[MHV-45980] Updated attachments screenreader alerts

- Added assurances so that screenreader reads remove attachment and attachment was successful first

- Cleanup aria-label file successfully attached alert so that it is read once

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-#59491

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-#59491

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-#59491

Commit:
Test fixes to Keyboard nav send Test

Fixed assertion script

mdewey commented on src/applications/mhv/secure-messaging/sass/secure-messaging.scss:
If its not in use, remove it. we can always look in the git history if we need to get it back.

mdewey commented on src/applications/mhv/secure-messaging/components/AttachmentsList.jsx:
Not for this PR, but for a future state, we might consider using a custom hook for the attachment logic.

mdewey reviewed (approved):
LGTM, minor feedback around refactoring.

Commit:
[MHV-45980] Cleanup

- Logic fix on bug in Reply messages, added clause for undefined attachments prop

- cleanup

Commit:
Merge branch 'main' into 45980-Remove-Attachment-Modal-59491

Commit:
[MHV-45980] Started unit tests for RemoveAttachmentsModal

Commit:
[MHV-45980] Updated unit tests to remove falsey results

alex-morgun-1 committed:
unit test - consolidate into const
